tc/core/polyhedral/cuda/mapped_scop.cc

```diff
@@ -506,7 +506,10 @@ isl::multi_union_pw_aff MappedScop::threadMappingSchedule(
     ids.emplace_back(mapping::ThreadId::makeId(i));
   }
   auto tupleId = isl::id(tree->ctx_, kBlock);
-  return extractDomainToIds(tree, ids, tupleId);
+  auto mapping = extractDomainToIds(tree, ids, tupleId);
+  // Check that all active elements got mapped.
+  CHECK(activeDomainPoints(schedule(), tree).is_subset(mapping.domain()));
```

Comment:
Shouldn't this check be put in `extractDomainToIds` intead?

I know you say

> The check was originally introduced in extractDomainToThreads and was
> meant to ensure that all branches in the subtree are indeed mapped to
> threads.  Perform the check inside threadMappingSchedule (successor of
> extractDomainToThreads) instead by checking that all active domain
> points are present in the mapping.

but I don't think that's the case.  The check you removed doesn't check that all branches
are mapped, only that every mapping node found maps to the requested ids.
So, the check you introduce here may be useful, but it's a separate issue and
looks to me like it belongs to `extractDomainToIds`.

Reply:
+ 1 for putting it inside `extractDomainToIds` and combine with the previously suggested checks).

Reply:
That function could be used precisely to check what is _not_ mapped by subtracting its result from the active points.  But i'll change for this PR.

Reply:
This is slightly annoying because it requires "extractDomainToIds" to take the root pointer which it does need otherwise.

Reply:
> This is slightly annoying because it requires "extractDomainToIds" to take the root poin
ter which it does need otherwise.

What was the motivation for extracting out extractDomainToIds
in the first place?  The commit message says it could be used
for other types of mapping ids, but this PR doesn't do that.
It's then difficult to reason about what extractDomainToIds
should look like.

skimo

Reply:
to have the `blockMappingSchedule` in the same way, it's now added